src/ray/object_manager/common.cc

```diff
@@ -134,7 +134,7 @@ Status PlasmaObjectHeader::WriteAcquire(
   RAY_CHECK(sem.header_sem);
 
   RAY_RETURN_NOT_OK(TryToAcquireSemaphore(sem.object_sem, timeout_point));
-  RAY_RETURN_NOT_OK(TryToAcquireSemaphore(sem.header_sem, timeout_point));
+  RAY_RETURN_NOT_OK(TryToAcquireSemaphore(sem.header_sem));
```

Comment:
Can we have a unit test that fails without this change but passes with this change?

Reply:
it is a bit difficult to write unit tests (I could verify it fixed issues before with bufferring PR when I used a different approach). Basically, there should be many downstream readers trying to acquire lock while a writer is trying to wrtie with timeout = 0. Let me try out